data/txHandler.go

```diff
@@ -834,52 +855,33 @@ func (handler *TxHandler) validateIncomingTxMessage(rawmsg network.IncomingMessa
 	wi := &txBacklogMsg{
 		rawmsg:                &rawmsg,
 		unverifiedTxGroup:     unverifiedTxGroup,
-		rawmsgDataHash:        msgKey,
 		unverifiedTxGroupHash: canonicalKey,
 		capguard:              nil,
+		syncCh:                make(chan network.ForwardingPolicy, 1),
 	}
 
-	if handler.checkAlreadyCommitted(wi) {
-		transactionMessagesAlreadyCommitted.Inc(nil)
-		return network.OutgoingMessage{
-			Action: network.Ignore,
-		}
-	}
-
-	err := handler.batchVerifier.Verify(wi.unverifiedTxGroup)
-	if err != nil {
-		handler.postProcessReportErrors(err)
-		logging.Base().Warnf("Received a malformed tx group %v: %v", wi.unverifiedTxGroup, err)
-		return network.OutgoingMessage{
-			Action: network.Disconnect,
-		}
-	}
-	verifiedTxGroup := wi.unverifiedTxGroup
-
-	// save the transaction, if it has high enough fee and not already in the cache
-	err = handler.txPool.Remember(verifiedTxGroup)
-	if err != nil {
-		handler.rememberReportErrors(err)
-		logging.Base().Debugf("could not remember tx: %v", err)
-		return network.OutgoingMessage{
-			Action: network.Ignore,
-		}
-	}
+	var action network.ForwardingPolicy
+	select {
+	case handler.backlogQueue <- wi:
+		action = <-wi.syncCh
```

Comment:
This is the real meat of the PR then?  You want to be able to treat the backlog handling go routine as a synchronous thing?  Could we put out the synchronous code of the backlog handler and call it here instead?

Reply:
we basically had the synchronous code of the backlog here (removed) and it caused locks congestion.